### PR TITLE
AL1 Configurator fix: misplaced arrow keys

### PR DESCRIPTION
**Pull request by noroadsleft**
re: qmk/qmk_configurator#186

The arrow keys in the JSON file were in the wrong places for both matrices.

### CODE REVIEW

**Comment by mechmerlin** [comment]
you can add in the RGB stuff by just changing the rules.mk sections to `yes` instead of `no`, you may need to disable mouse keys to get the hex size to fit though.

File: keyboards/al1/info.json [hunk 1]
```diff
@@ -5,12 +5,11 @@
...
-      }
+    "LAYOUT": {
```
Comment (mechmerlin) [inline keyboards/al1/info.json]: did you just make an entirely new info.json? It's hard to see what exactly changed here.

**Comment by drashna** [comment]
@mechmerlin file size appears to be fine.  No warnings in travis about it, at least.

Also, turning off COMMAND is probably a better way to save space.

File: keyboards/al1/info.json (hunk 1 above)
Comment (noroadsleft) [reply keyboards/al1/info.json]: Used a [JSON formatter](https://jsonformatter.org/) to get a tree view, and rearranged the objects into the proper order, then compressed it for white space (it bothers me when the JSON files have a bunch of lines, even though it has no effect on functionality).

I have to leave soon but I'll edit `rules.mk` and put another comment explaining how exactly I did it after I get back.

**Comment by noroadsleft** [comment]
@mechmerlin re: the JSON formatting

The way the Configurator uses the JSON files is as follows:

The Configurator API reads the layout macro from the keyboard's `.h` and assigns the JSON objects for the key in order, left to right, top to bottom.

`keyboard.h`:

```c
#define LAYOUT( \
    k00, k01, k02, \
    k10, k11, k12  \
) { \
    { k00, k01, k02 }, \
    { k10, k11, k12 }, \
}
```

Let's say the [key at k11 is offset down 0.5u from k10 and k12](http://www.keyboard-layout-editor.com/##@@=k00&=k01&=k02%3B&@=k10&_x:1%3B&=k12%3B&@_y:-0.5&x:1%3B&=k11). Because of how the QMK KLE-JSON converter works, it sorts all the keys based on x- and y-axis positions. Because k11 is offset, it's `y` value is higher than that of k10 and k12, so the converter treats k11 as if it's after k10 and k12.

Putting the above layout through the converter results in the following JSON (edited only for readability):

```json
{
    "keyboard_name": "", 
    "url": "", 
    "maintainer": "qmk", 
    "bootloader": "", 
    "width": 3, 
    "height": 2.5, 
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"label":"k00", "x":0, "y":0},
                {"label":"k01", "x":1, "y":0},
                {"label":"k02", "x":2, "y":0},
                {"label":"k10", "x":0, "y":1},
                {"label":"k12", "x":2, "y":1},
                {"label":"k11", "x":1, "y":1.5}
            ]
        }
    }
}
```

K11 is last, because it has the highest y-value at 1.5. To have it tie correctly to the layout matrix macro, it needs to be second-to-last:

```json
{
    "keyboard_name": "", 
    "url": "", 
    "maintainer": "qmk", 
    "bootloader": "", 
    "width": 3, 
    "height": 2.5, 
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"label":"k00", "x":0, "y":0},
                {"label":"k01", "x":1, "y":0},
                {"label":"k02", "x":2, "y":0},
                {"label":"k10", "x":0, "y":1},
                {"label":"k11", "x":1, "y":1.5},
                {"label":"k12", "x":2, "y":1}
            ]
        }
    }
}
```

To work around this, if I'm dealing with a keyboard where keys are offset vertically, I position them in KLE as if they are not offset, run the conversion, and then edit the y-value by hand in the resulting JSON.

**Comment by noroadsleft** [comment]
@mechmerlin does this PCB support RGB? I looked at the GH GB page and there's no mention of it, and I can't see any place for RGB LEDs. Trying to fix a compile error.

```make
Making al1 with keymap default                                                                         [ERRORS]
In file included from quantum/quantum.h:34:0,
                 from keyboards/al1/al1.h:19,
                 from keyboards/al1/al1.c:16:
quantum/rgblight.h:83:21: error: 'RGBLED_NUM' undeclared here (not in a function)
 extern LED_TYPE led[RGBLED_NUM];
                     ^
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:360: .build/obj_al1_default/keyboards/al1/al1.o] Error 1
```

Both keymaps have the same error.

**Comment by drashna** [comment]
Is this ready to merge?

**Comment by noroadsleft** [comment]
Should be ready now, as long as Travis passes.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**